File: data-r2dbc/src/test/groovy/io/micronaut/data/r2dbc/postgres/http/SampleEntity.java
package io.micronaut.data.r2dbc.postgres.http;

import io.micronaut.core.annotation.Introspected;
import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;

import java.io.Serializable;

/**
 * Represents sample.
 * 
 * @author seprokof
 *
 */
@Introspected
@MappedEntity(value = "pg_sample", alias = "sample")
public class SampleEntity implements Serializable {
    private static final long serialVersionUID = -5410891143601576439L;

    @Id
    @GeneratedValue
    private Long id;
    private String data;

    public SampleEntity() {
    }

    public SampleEntity(Long id, String data) {
        this.id = id;
        this.data = data;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getData() {
        return data;
    }

    public void setData(String data) {
        this.data = data;
    }
}

File: doc-examples/jdbc-example-kotlin/src/main/kotlin/example/SampleEntity.kt
package example

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import io.micronaut.data.annotation.Relation

@MappedEntity
data class SampleEntity(
    @field:Id
    val id: Long,

    val name: String? = null,

    @GeneratedValue
    val example: String? = null,

    @Relation(value = Relation.Kind.EMBEDDED)
    val part: Part = Part()
)
